low-level/files/opt/seagate/sspl/setup/setup_yum_repos.py

```diff
@@ -0,0 +1,187 @@
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+
+# *********************************************************
+#  Description:   Add common and sspl repos to yum.repos.d
+# *********************************************************
+
+import os
+import sys
+import shutil
+import subprocess
+
+CORTX_BASE_URL = "http://cortx-storage.colo.seagate.com/releases/cortx"
+
+
+class SetupYumRepo:
+    """Create repo files under /etc/yum.repos.d"""
+
+    name = "Setup Yum Repo"
+
+    def __init__(self, target_build_url):
+        """Initialize yum repo object."""
+        self.build_url = target_build_url
+        self.cortx_deps_repo = None
+        self.epel_repo = None
+        self.url_sspl_repo = None
+        self.url_uploads_repo = None
+        self.cortx_commons_repo = "/etc/yum.repos.d/cortx_commons.repo"
+        self.cortx_platform_base_repo = "/etc/yum.repos.d/cortx_platform_base.repo"
+        self.cortx_platform_extras_repo = "/etc/yum.repos.d/cortx_platform_extras.repo"
+        self.third_party_epel_repo = "/etc/yum.repos.d/3rd_party_epel.repo"
+        self.sspl_repo = "/etc/yum.repos.d/sspl.repo"
+        self.sspl_uploads_repo = "/etc/yum.repos.d/sspl_uploads.repo"
+
+    def _validate_centos_release_support(self):
+        """Get CORTX url based on centos release."""
+        file = "/etc/centos-release"
+        with open(file) as fObj:
+            content = fObj.read()
+        if "CentOS Linux release 7.8" in content:
+            self.url_local_repo_commons="%s/third-party-deps/centos/centos-7.8.2003/" % (CORTX_BASE_URL)
+            self.url_uploads_repo = "%s/uploads/centos/centos-7.8.2003/" % (CORTX_BASE_URL)
+        elif "CentOS Linux release 7.7" in content:
+            self.url_local_repo_commons="%s/third-party-deps/centos/centos-7.7.1908/" % (CORTX_BASE_URL)
+            self.url_uploads_repo="%s/uploads/centos/centos-7.7.1908/" % (CORTX_BASE_URL)
+        else:
+            raise Exception("%s: %s" % (self.name,
+                                        "OS version not supported. " +
+                                        "Supported OS versions are CentOS-7.7 and CentOS-7.8"))
+
+    def set_repo_url(self):
+        """Make build specific url."""
+        if self.build_url:
+            self.cortx_deps_repo = "%s/3rd_party" % self.build_url
+            self.epel_repo = "%s/EPEL-7" % self.build_url
+            self.url_local_repo_commons = self.cortx_deps_repo
+            self.url_sspl_repo = "%s/cortx_iso" % self.build_url
+        self._validate_centos_release_support()
+
+    def create_commons_repos(self):
+        """Create common platform base, extra and epel repos."""
+        # Create common repo
+        content = """[cortx_commons]
+name=cortx_commons
```

Comment:
why the indentation not maintained? same down below all over this script

Reply:
Its by VS code editor plugin which auto aligns code. 
I have corrected it now.